sql/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java

```diff
@@ -283,6 +294,36 @@ public void update(ClusterChangedEvent event) {
                 }
             }
         }
+
+        // re register UDFs for this schema
+        functions.deregisterSchemaFunctions(schemaName);
+        UserDefinedFunctionsMetaData udfMetaData = newMetaData.custom(UserDefinedFunctionsMetaData.TYPE);
+        if (udfMetaData != null) {
+            List<UserDefinedFunctionMetaData> udfFunctions = udfMetaData.functionsMetaData().stream()
+                .filter(function -> schemaName.equals(function.schema()))
+                .collect(Collectors.toList());
+            functions.registerSchemaFunctionResolvers(schemaName, toFunctionImpl(udfFunctions, logger));
+        }
+    }
+
+    @VisibleForTesting
+    Map<FunctionIdent, FunctionImplementation> toFunctionImpl(List<UserDefinedFunctionMetaData> functionsMetadata,
+                                                              Logger logger) {
+        Map<FunctionIdent, FunctionImplementation> udfFunctions = new HashMap<>();
+        for (UserDefinedFunctionMetaData function : functionsMetadata) {
+            try {
+                UDFLanguage lang = udfService.getLanguage(function.language());
```

Comment:
not sure if I like how the responsibilities are split. What if we just had a `udfService.updateFunctionRegistry(filteredFunctionsMetaData)` that did all this, leaving only the filtering of the metadata to the schema?